woocommerce/woocommerce-ios
# EU-Bound Shipping Notice: Create EU Customs scenario validation

Closes #9593 and #9594 

Why
==========
The objective of the EU-Bound Shipping Notice is to warn the user whenever a Shipping Label is under creation where the origin is the US and the destination is one of the EU countries inside the new Customs rules. Following the USPS instructions, the notice will be triggered when the origin address is from the US and the destination address is one of the following countries:

<img width="697" alt="image" src="https://user-images.githubusercontent.com/5920403/235547782-54e2b1de-879a-4c8e-9568-1e9e2fa8d03d.png">

How
==========
Implements the `EUCustomsScenarioValidator` that checks if the origin and destination country codes are included in the validation list and an exhaustive unit test coverage to ensure that all required scenarios are always there.

Screen Capture
==========
https://github.com/woocommerce/woocommerce-ios/assets/5920403/acf4aa09-8f45-4daa-bc3e-907ac0740f8a
  
How to Test
==========
For better testing the scenarios introduced here, I recommend setting the `AppSettingsStore.loadEUShippingNoticeDismissState` to always trigger `onCompletion(.success(false))`.

1. Open the app using a store with the `Shipping Labels` plugin activated
2. Open the Order Details of an order with the `Processing` status
3. Hit the `Create Shipping Label` button
4. Verify that the Banner only appears when the Origin country is `US` and the Destination country is one of the EU countries listed above.
5. Verify that the `Customs` view also follows the exact same visibility configuration from the `Create Shipping Label` view

Update release notes:
